test/common/http/http2/codec_impl_test.cc

```diff
@@ -608,6 +608,7 @@ TEST_P(Http2CodecImplTest, ContinueHeaders) {
   driveToCompletion();
 };
 
+#if !defined(ENVOY_ENABLE_UHV)
```

Comment:
I would add a function to this test file 

```
void skipTestForUhv() {
#ifdef ENVOY_ENABLE_UHV
  GTEST_SKIP();
#endif
}
```
And then just call it after the initialize() call, to skip tests under UHV. This will help avoid large scale ifdefing across files.

Reply:
Once the nghttp2 setting has been applied, entering the test is enough to cause the codec to break due to the various ASSERTs, so the only way to actually skip those tests is to not compile them

Reply:
> Once the nghttp2 setting has been applied, entering the test is enough to cause the codec to break due to the various ASSERTs, so the only way to actually skip those tests is to not compile them

@yanavlasov should we keep disabling the entire tests with an `#if !defined(ENVOY_ENABLE_UHV)`, or should approach things differently? Right now I was trying to use the `skipTestForUhv` approach by adding the missing callbacks but I'm breaking even tests